core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala

```diff
@@ -141,25 +144,51 @@ abstract class KafkaServerTestHarness extends QuorumTestHarness {
     createBrokers(startup)
   }
 
+  def createOffsetsTopic(
+    listenerName: ListenerName = listenerName,
+    adminClientConfig: Properties = new Properties
+  ): Unit = {
+    if (isKRaftTest()) {
+      TestUtils.withAdmin(brokers, listenerName, adminClientConfig) { admin =>
```

Comment:
How about removing `withAdmin` and manually composing this with `resource` and `createAdminClient` in `TestUtils`?